File: src/lib/validations/driver.ts
import { z } from 'zod'

export const driverSchema = z.object({
  firstName: z.string().min(1, 'First name is required').max(200),
  lastName: z.string().min(1, 'Last name is required').max(200),
  email: z.string().max(254).email('Invalid email').optional().or(z.literal('')),
  phone: z.string().max(30).optional().or(z.literal('')),
  address: z.string().max(500).optional().or(z.literal('')),
  city: z.string().max(500).optional().or(z.literal('')),
  state: z.string().max(2).optional().or(z.literal('')),
  zip: z.string().max(20).optional().or(z.literal('')),
  licenseNumber: z.string().max(200).optional().or(z.literal('')),
  driverType: z.enum(['company', 'owner_operator']).default('company'),
  driverStatus: z.enum(['active', 'inactive']).default('active'),
  payType: z.enum(['percentage_of_carrier_pay', 'dispatch_fee_percent', 'per_mile', 'per_car']).default('percentage_of_carrier_pay'),
  payRate: z.coerce.number().min(0, 'Pay rate must be 0 or more').max(100, 'Pay rate cannot exceed 100').default(0),
  notes: z.string().max(5000).optional().or(z.literal('')),
})

export type DriverFormValues = z.infer<typeof driverSchema>
export type DriverFormInput = z.input<typeof driverSchema>

File: VroomXDriver/Models/Driver.swift
import Foundation

/// Matches VroomX `drivers` table (migrations 00002 + 00006).
/// Uses `VroomXDriver` name to avoid collision with the app module name.
struct VroomXDriverModel: Codable, Identifiable {
    let id: String
    let tenantId: String
    let firstName: String
    let lastName: String
    let email: String?
    let phone: String?
    let address: String?
    let city: String?
    let state: String?
    let zip: String?
    let licenseNumber: String?
    let driverType: DriverType
    let driverStatus: String
    let payType: DriverPayType
    let payRate: Double
    let authUserId: String?
    let pinHash: String?
    let notes: String?
    let createdAt: String
    let updatedAt: String

    // MARK: - Computed Properties

    var fullName: String {
        "\(firstName) \(lastName)"
    }

    var initials: String {
        let first = firstName.prefix(1).uppercased()
        let last = lastName.prefix(1).uppercased()
        return "\(first)\(last)"
    }

    var isActive: Bool {
        driverStatus == "active"
    }

    // MARK: - CodingKeys

    enum CodingKeys: String, CodingKey {
        case id
        case tenantId = "tenant_id"
        case firstName = "first_name"
        case lastName = "last_name"
        case email
        case phone
        case address
        case city
        case state
        case zip
        case licenseNumber = "license_number"
        case driverType = "driver_type"
        case driverStatus = "driver_status"
        case payType = "pay_type"
        case payRate = "pay_rate"
        case authUserId = "auth_user_id"
        case pinHash = "pin_hash"
        case notes
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}
